### PR TITLE
samples: matter: Updated Matter version to pull bug fixes

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
Updated Matter version to pull bug fixes:
* nRF53 SED hardfault bug, caused by QSPI uninitialized state - KRKNWK-14090
* QR code url and NFC tag generation bug, caused by invalid QR code buffer size. - KRKNWK-14100, KRKNWK-14101

Signed-off-by: Kamil Kasperczyk <kamil.kasperczyk@nordicsemi.no>

### CODE REVIEW

File: samples/matter/common/src/ota_util.cpp
Comment (Damian-Nordic) [inline samples/matter/common/src/ota_util.cpp:33]: It's duplicate of `GetOTAImageProcessor` function below. Remove either this or that one.

**Comment by Damian-Nordic** [comment]
Could you extend the description why this helps? If the issue is that QSPI is not initialized when starting the DFU why putting it to sleep helps?

**Comment by kkasperczyk-no** [comment]
> Could you extend the description why this helps? If the issue is that QSPI is not initialized when starting the DFU why putting it to sleep helps?

It was put to sleep in the `OTAImageProcessorImplPMDevice` constructor and then the QSPI state was uinitialized during OTARequestor init. I just moved putting it to sleep from constructor to the additional method called on the end of OTARequestor init.

**Review by markaj-nordic — COMMENTED** [review]
Maybe we can consider the following approach with a minimal interface extension:
- remove DoAction(ExtFlashHandler::Action::SLEEP) call from OTAImageProcessorImplPMDevice c'tor
- add default virtual PutPMDeviceSleep() with empty body to the OTAImageProcessorImpl and
- overload it with proper implementation in OTAImageProcessorImplPMDevice
- call the imageProcessor.PutPMDeviceSleep() at the end of InitBasicOTARequestor()

File: samples/matter/common/src/ota_util.cpp
Comment (markaj-nordic) [inline samples/matter/common/src/ota_util.cpp:51]: Where is the `PutPMDeviceSleep` defined? I cannot find it anywhere in the Github diff of this commit :)

Comment (kkasperczyk-no) [reply samples/matter/common/src/ota_util.cpp:51]: It is in the corresponding commit to the sdk-connectedhomeip fork: https://github.com/nrfconnect/sdk-connectedhomeip/pull/65/files#diff-58afcbe9511e695bb69ec55703ca86add9bd5c890fac32ce61cdcb1f103f2782R223

**Comment by kkasperczyk-no** [comment]
>  add default virtual PutPMDeviceSleep() with empty body to the OTAImageProcessorImpl and overload it with proper implementation in OTAImageProcessorImplPMDevice

It doesn't feel right to me to add `PutPMDeviceSleep` to the `OTAImageProcessorImpl`, as `OTAImageProcessorImpl` doesn't need to know what is PMDevice.

**Comment by markaj-nordic** [comment]
> > add default virtual PutPMDeviceSleep() with empty body to the OTAImageProcessorImpl and overload it with proper implementation in OTAImageProcessorImplPMDevice
> 
> It doesn't feel right to me to add `PutPMDeviceSleep` to the `OTAImageProcessorImpl`, as `OTAImageProcessorImpl` doesn't need to know what is PMDevice.

@kkasperczyk-no I got your point. But what about changing the name to something like "EnterSleepMode"? I mean, the `OTAImageProcessorImpl` is the implementation of abstract OTAProcessor tailored to our platform, so I believe there is nothing wrong in extending the underlying interface to match our needs. I just feel the existing approach is more neat :) @Damian-Nordic  what do you think?

**Comment by Damian-Nordic** [comment]
@markaj-nordic I think we can talk about that after the tomorrow's standup as I'm not sure if I fully understand the issue. In particular why it only occurs on nRF53.

**Comment by kkasperczyk-no** [comment]
So what is the conclusion on that? :) Are we ok with the current implementation or need to go with some changes?

**Review by markaj-nordic — APPROVED** [review]

**Review by Damian-Nordic — APPROVED** [review]

File: samples/matter/common/src/dfu_over_smp.cpp
Comment (Damian-Nordic) [inline samples/matter/common/src/dfu_over_smp.cpp:42]: It is a bit weird that we call `FlashWakeUp` on `dfu_stopped_cb`. Isn't dfu_started_cb enough?

Comment (markaj-nordic) [reply samples/matter/common/src/dfu_over_smp.cpp:42]: I guess we should rather force it to sleep instead of waking up when the DFU is stopped. Thanks for the catch.

**Comment by mbolivar-nordic** [comment]
https://github.com/nrfconnect/sdk-connectedhomeip/pull/65 is merged; please update west.yml to the final SHA

**Comment by greg-fer** [comment]
Please wait with merges of ``doc-required`` PRs until an approval from tech writer. Thanks.

**Pull request merged**
